src/components/api/kvstore_itf.h

```diff
@@ -396,6 +396,29 @@ class IKVStore : public Component::IBase
     return E_NOT_SUPPORTED;
   }
 
+  /**
+   * Allocate memory for zero copy DMA
+   *
+   * @param vaddr [out] allocated memory buffer
+   * @param len [in] length of memory buffer in bytes
+   *
+   * @return Memory handle or NULL on not supported.
+   */
+  virtual memory_handle_t allocate_direct_memory(void * &vaddr, size_t len){
```

Comment:
This is not actually necessary for other stores,  but nvmestore does requires special aligned direct memory.